### PR TITLE
Add bcftools reheader (#585)

### PR DESCRIPTION
**Pull request by bjohnnyd**
Add `bcftools/reheader` module.  One thing that needs working on is the test for adding a new header which either requires a new test dataset or can be obtained with `bcftools/view` if that module is created as well. 

@drpatelh  if you have a preference on the two options let me know and I can add the dataset or create a `bcftools/view` module.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/bcftools/reheader/meta.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      description: VCF/BCF file
+      pattern: "*.{vcf.gz,vcf,bcf}"
+  - fai:
+      type: file
+      description: Fasta index to update header sequences with
+      pattern: "*.{fai}"
+  - header:
+      type: file
+      description: New header to add to the VCF
+      pattern: "*.{hr}"
```
Comment (bjohnnyd) [inline modules/bcftools/reheader/meta.yml:33]: I was not sure on this extension, technically this can be anything and maybe even `.txt` or `.vcf`?

Comment (drpatelh) [reply modules/bcftools/reheader/meta.yml:33]: I would probably call it `*header.txt`

File: tests/modules/bcftools/reheader/main.nf [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+workflow test_bcftools_reheader_sequences {
+
+    input = [
+        [ id:'test', single_end:false ],
+        file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true)
+    ]
+    fai    = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
+    header = []
+
```
Comment (drpatelh) [inline tests/modules/bcftools/reheader/main.nf:15]:
```suggestion
```

File: modules/bcftools/reheader/meta.yml (hunk 1 above)
Comment (jfy133) [reply modules/bcftools/reheader/meta.yml:33]: Wouldn't just `.txt` be safer, if assuming the pipeline user will customise the output file with `$options.suffix`?

Comment (drpatelh) [reply modules/bcftools/reheader/meta.yml:33]: The problem here is that a `.txt` file could be anything and in theory could be overwritten by a similar named `.txt` file from another module which is why adding the `.header.txt` clarifies it's a file containing the header. This doesn't stop developers from adding their own custom `prefix` though e.g. `sample1.header.txt` / `sample1.filtered.header.txt` etc.

File: tests/modules/bcftools/reheader/main.nf [hunk 3]
```diff
@@ -13,6 +13,29 @@ workflow test_bcftools_reheader_sequences {
...
+}
+
+workflow test_bcftools_reheader_new_header_update_sequences {
+
+    input = [
+        [ id:'test', single_end:false ],
+        file(params.test_data['sarscov2']['illumina']['test_vcf_gz'], checkIfExists: true)
+    ]
+    fai    = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
+    header = file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
```
Comment (bjohnnyd) [inline tests/modules/bcftools/reheader/main.nf:38]: Actually I can pass anything for header. I went with a vcf  for now?

**Review by drpatelh — APPROVED** [review]
🍬

**Pull request merged**